leasing/models/land_use_agreement.py

```diff
@@ -695,3 +728,112 @@ def generate_number(self):
             self.save()
 
         return self.number
+
+    def update_amounts(self):
+        pass
+
+
+class LandUseAgreementInvoiceRow(TimeStampedSafeDeleteModel):
+    """
+    In Finnish: Rivi laskulla
+    """
+
+    # In Finnish: Lasku
+    invoice = models.ForeignKey(
+        LandUseAgreementInvoice,
+        verbose_name=_("Invoice"),
+        related_name="rows",
+        on_delete=models.CASCADE,
+    )
+
+    # In Finnish: Osapuoli
+    litigant = models.ForeignKey(
+        LandUseAgreementLitigant,
+        verbose_name=_("Litigant"),
+        related_name="+",
+        null=True,
+        blank=True,
+        on_delete=models.PROTECT,
+    )
+
+    # In Finnish: Saamislaji
+    receivable_type = models.ForeignKey(
+        LandUseAgreementReceivableType,
+        verbose_name=_("Receivable type"),
+        related_name="+",
+        on_delete=models.PROTECT,
+    )
+
+    # In Finnish: Laskutettava määrä
+    amount = models.DecimalField(
+        verbose_name=_("Amount"), max_digits=10, decimal_places=2, default=0
+    )
+
+    # In Finnish: Korvauksen määrä €
+    compensation_amount = models.DecimalField(
+        verbose_name=_("Compensation amount"),
+        decimal_places=2,
+        max_digits=12,
+        default=0,
+    )
+
+    # In Finnish: Selite
+    description = models.TextField(verbose_name=_("Description"), null=True, blank=True)
+
+    # In Finnish: Korotuksen määrä %
+    increase_percentage = models.DecimalField(
+        verbose_name=_("Increase percentage"),
+        decimal_places=2,
+        max_digits=12,
+        default=0,
+    )
+
+    # In Finnish: Asemakaavan lainvoimaisuuspvm
+    plan_lawfulness_date = models.DateField(
+        verbose_name=_("Plan lawfulness date"), null=True, blank=True
+    )
+
+    # In Finnish: Allekirjoituspvm
+    sign_date = models.DateField(verbose_name=_("Sign date"), null=True, blank=True)
+
+    recursive_get_related_skip_relations = ["invoice"]
+
+    class Meta:
+        verbose_name = pgettext_lazy("Model name", "Invoice row")
+        verbose_name_plural = pgettext_lazy("Model name", "Invoice rows")
+
+    def update_amount(self):
+        self.amount = 123
+
+
+class LandUseAgreementInvoicePayment(TimeStampedSafeDeleteModel):
+    """
+    In Finnish: Maksusuoritus
+    """
+
+    # In Finnish: Lasku
+    invoice = models.ForeignKey(
+        LandUseAgreementInvoice,
+        verbose_name=_("Invoice"),
+        related_name="payments",
+        on_delete=models.CASCADE,
+    )
+
+    # In Finnish: Maksettu määrä
+    paid_amount = models.DecimalField(
+        verbose_name=_("Paid amount"), max_digits=10, decimal_places=2
+    )
+
+    # In Finnish: Maksettu pvm
+    paid_date = models.DateField(verbose_name=_("Paid date"))
+
+    # In Finnish: Arkistointitunnus
+    filing_code = models.CharField(
+        verbose_name=_("Name"), null=True, blank=True, max_length=35
```

Comment:
Since using `null=True` on `CharField`s and `TextField`s should be avoided, it should be removed here. (https://docs.djangoproject.com/en/2.2/ref/models/fields/#django.db.models.Field.null).

Reply:
fixed